src/macros.rs

```diff
@@ -30,6 +30,23 @@ pub use time_macros::date;
 ///
 /// [`OffsetDateTime`]: crate::OffsetDateTime
 /// [`PrimitiveDateTime`]: crate::PrimitiveDateTime
+///
+/// ```rust
+/// # use time::{Date, Time, Month, macros::datetime, PrimitiveDateTime, UtcOffset};
+/// assert_eq!(
+///     datetime!(2020-01-01 0:00),
+///     PrimitiveDateTime::new(Date::from_calendar_date(2020, Month::January, 1)?, Time::MIDNIGHT)
```

Comment:
```suggestion
///     Date::from_calendar_date(2020, Month::January, 1)?.midnight()
```

Same for the other two. Other than this nit, LGTM.